.emacs.d/config.org

```diff
@@ -55,62 +55,392 @@
-                use-package-always-demand t))
+   (eval-when-compile
+     (require 'use-package))
 
-(setq-default byte-compile-warnings nil)
+   ;; Yes, it's a bit meta...
+   (use-package use-package
+     :config
+     (setq-default use-package-always-defer t
+                   use-package-always-demand nil
+                   use-package-always-ensure t
+                   use-package-verbose t))
 
-(use-package use-package-ensure-system-package)
+   (setq-default byte-compile-warnings nil)
    #+end_src
+
+* Projects
+
+  - https://github.com/bbatsov/projectile
+  - https://www.projectile.mx/en/latest/
+
+  Projectile gives Emacs project management features, such as navigating files, finding occurences of a string, running build commands and more.
+  It detects project folders through version control (git, svn, ...) and you can optionally give a default project folder path within the config.
+
+  #+begin_src emacs-lisp
+  (use-package projectile
+    ;; Getting complaints about undefined variables when these are placed in
+    ;; `:custom`.
+    :config
+    ;; Useful for any CMake-based project, and using `ccls` with C/C++
+    (setq projectile-project-root-files-top-down-recurring
+     (append
+      '("compile_commands.json" ".ccls")
+      projectile-project-root-files-top-down-recurring))
+    (setq projectile-globally-ignored-directories
+     (append
+      '("build" "CMakeFiles" ".ccls-cache")
+      projectile-globally-ignored-directories))
+    (setq projectile-globally-ignored-files
+     (append
+      '("cmake_install.cmake")
+      projectile-globally-ignored-files))
+    :custom
+    (projectile-project-search-path '("~/Projects/"))
+    (projectile-enable-caching t)
+    (projectile-require-project-root t)
+    (projectile-sort-order 'access-time)
+    :init
+    (projectile-mode))
+  #+end_src
+
+* Magit
+
+  [[https://github.com/magit/magit][Magit]] is a Git interface, just like you have in many IDEs and text editors, but combining that with all the power of using Git from the command line.
+
+  #+begin_src emacs-lisp
+  (use-package magit)
+  #+end_src
+
+* Key bindings
+
+  [[https://github.com/justbur/emacs-which-key][Which-key]] is a package that displays what bindings are available when you start pressing a key. It is incredibly useful, especially when you are new to Emacs or when you are trying a new package.
+
+  #+BEGIN_SRC emacs-lisp
+  (use-package which-key
+    :init
+    (which-key-mode))
+  #+END_SRC
+
+* Linters
+
+  The two main frameworks that allow Emacs to interface with external linters are
+  - [[https://www.gnu.org/software/emacs/manual/html_node/emacs/Flymake.html][flymake]]
+  - [[https://www.flycheck.org/en/latest/][flycheck]]
+  A detailed comparison is at https://www.flycheck.org/en/latest/user/flycheck-versus-flymake.html. We choose to use flycheck due to the sheer number of extensions written for it; pick your favorite language linter, and there’s probably a flycheck extension for it.
+
```

Comment:
We may want to add explicit disclaimers that Windows isn't supported as shown [here](https://www.flycheck.org/en/latest/#try-out).